src/server/controllers/user.js

```diff
@@ -6,31 +6,53 @@ const prisma = new PrismaClient();
 const jwtSecret = 'mysecret';
 
 const register = async (req, res) => {
-    const { username, password } = req.body;
+    const saltRounds = 10;
 
-    const createdUser = null;
+    const { username, password } = req.body;
 
-    res.json({ data: createdUser });
+    try {
+        const hashedPassword = await bcrypt.hash(password, saltRounds);
+        const newUser = await prisma.user.create({
+            data: {
+                username,
+                password: hashedPassword,
+            },
+        });
+
+        return res.status(201).json({ message: 'User created successfully.', userId: newUser.id });
+    } catch (error) {
+        return res.status(500).json({ error: 'Error creating new user.' });
```

Comment:
Would be nice to check the error to narrow it down to what went wrong. It may not be a 500 error. For example, if the error is that the username has already been used, it would be a 409 (conflict) error